fenwick/src/main/java/org/opencadc/fenwick/StorageLocationEventSync.java

```diff
@@ -0,0 +1,234 @@
+*  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+*  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+*  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+*  General Public License for           Générale Publique GNU Affero
+*  more details.                        pour plus de détails.
+*
+*  You should have received             Vous devriez avoir reçu une
+*  a copy of the GNU Affero             copie de la Licence Générale
+*  General Public License along         Publique GNU Affero avec
+*  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+*  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+*                                       <http://www.gnu.org/licenses/>.
+*
+************************************************************************
+*/
+
+package org.opencadc.fenwick;
+
+import ca.nrc.cadc.date.DateUtil;
+import ca.nrc.cadc.db.TransactionManager;
+import ca.nrc.cadc.io.ResourceIterator;
+import ca.nrc.cadc.net.ResourceNotFoundException;
+import ca.nrc.cadc.net.TransientException;
+import java.io.IOException;
+import java.net.URI;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.text.DateFormat;
+import java.util.Date;
+import org.apache.log4j.Logger;
+import org.opencadc.inventory.Artifact;
+import org.opencadc.inventory.SiteLocation;
+import org.opencadc.inventory.StorageLocationEvent;
+import org.opencadc.inventory.StorageSite;
+import org.opencadc.inventory.db.ArtifactDAO;
+import org.opencadc.inventory.db.HarvestState;
+import org.opencadc.inventory.query.StorageLocationEventRowMapper;
+import org.opencadc.tap.TapClient;
+
+/**
+ *
+ * @author pdowler
+ */
+public class StorageLocationEventSync extends AbstractSync {
+    private static final Logger log = Logger.getLogger(StorageLocationEventSync.class);
+
+    private final StorageSite storageSite;
+    private final TapClient<StorageLocationEvent> tapClient;
+    
+    public StorageLocationEventSync(ArtifactDAO artifactDAO, URI resourceID, 
+            int querySleepInterval, int maxRetryInterval, 
+            StorageSite storageSite) {
+        super(artifactDAO, resourceID, querySleepInterval, maxRetryInterval);
+        this.storageSite = storageSite;
+        try {
+            this.tapClient = new TapClient<>(resourceID);
+        } catch (ResourceNotFoundException ex) {
+            throw new IllegalArgumentException("invalid config: query service not found: " + resourceID);
+        }
+    }
+    
+    @Override
+    void doit() throws ResourceNotFoundException, IOException, IllegalStateException, TransientException, InterruptedException {
+        final MessageDigest messageDigest;
+        try {
+            messageDigest = MessageDigest.getInstance("MD5");
+        } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException("BUG: failed to get instance of MD5", e);
+        }
```

Comment:
messageDisgest isn't used.

Reply:
Oh, actually it should be used to check that the event is intact... and so should the two Deleted??Sync classes. 
That was missing from day one (and I reviewed that code :-) Adding those checks.